### PR TITLE
[move][stdlib] Add vector::split_off and vector::replace library methods

### PR DESCRIPTION
**Pull request by igor-aptos**
## Description
When we store un-drop-able objects in a vector, handling changes to vector needs to be done without dropping any elements in the process (or creating temporary ones).

Adding two missing utilities to help with that:
* split_off - which moves part of the vector into a new vector
* replace - which replaces value at particular index. 

Both could have more efficient implementations, if implemented in rust. But seems like that is the case for many functions in vector.move, so starting with straightforward implementation.

## How Has This Been Tested?
provided unit tests

## Type of Change
- [x] New feature

## Which Components or Systems Does This Change Impact?
- [x] Aptos Framework

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by msmouse** [comment]
in the summary: "remove - which replaces value at particular index." : "remove" should be "replace" 😂

File: aptos-move/framework/move-stdlib/sources/vector.move [hunk 1]
```diff
@@ -122,6 +122,22 @@ module std::vector {
         pragma intrinsic = true;
     }
 
+    /// Splits the collection into two at the given index.
+    /// Returns a newly allocated vector containing the elements in the range [at, len).
+    /// After the call, the original vector will be left containing the elements [0, at)
+    /// with its previous capacity unchanged.
+    public fun split_off<Element>(self: &mut vector<Element>, at: u64): vector<Element> {
```
Comment (lightmark) [inline aptos-move/framework/move-stdlib/sources/vector.move:129]: Do you think it's more efficient with your `vector::range` copy and a native `vector::resize`?

Comment (lightmark) [inline aptos-move/framework/move-stdlib/sources/vector.move:129]: what's the diff with `trim`?

Comment (lightmark) [reply aptos-move/framework/move-stdlib/sources/vector.move:129]: I believe it's better to make trim native tho

Comment (lightmark) [reply aptos-move/framework/move-stdlib/sources/vector.move:129]: the move vector seems not to have the concept of `capacity`... 
idk who added this... but `push_back` one by one is not that efficient? :)

Comment (igor-aptos) [reply aptos-move/framework/move-stdlib/sources/vector.move:129]: I did that in #14862, let me just merge this into that PR.

**Comment by igor-aptos** [comment]
merged into #14862

**Pull request closed without merging**